src/loading_ui.cpp

```diff
@@ -92,7 +92,18 @@ static void update_state( const std::string &context, const std::string &step )
 #ifdef TILES
         cata_path path = PATH_INFO::gfxdir() / "cdda.png";
         SDL_Surface_Ptr surf = load_image( path.get_unrelative_path().u8string().c_str() );
-        gLUI->splash_size = { static_cast<float>( surf->w ), static_cast<float>( surf->h ) };
+        // leave some space for the loading text...
+        ImVec2 max_size = ImGui::GetMainViewport()->Size * 0.9;
+        // since drawing the loading text depends on the width of the loading image(?!), enforce a minimum size
```

Comment:
The text is simply centered relative to the image. It could just as easily be centered relative to the viewport or the window, but I‌ already had the splash size ready to hand so I‌ just wrote it that way. And you don’t actually use `min_size` anyway.

Reply:
(Sorry for the *extremely* late reply, I really was gaming hard with Satisfactory!)

It actually cuts off any text that is outside of the image's width. It is not merely centered, but just plain gone. Actually it seems it's cutting off some pixels on the right-hand side too? This is a roughly 100x100 example loading screen, with a white box around the edge to make it easy to identify.

(This image is from master, without this PR applied)
![image](https://github.com/user-attachments/assets/f9b3c78b-75e1-425a-b32b-ab710782dd0f)

Scaling it both "up" (if below maximum available space) and "down" should resolve both of these issues. Though I'm not sure why the text is like this - it doesn't seem to be how you described it and I don't think it's what you intended.

min_size was just left over from an earlier approach, I removed it now.

Reply:
>I'm not sure why the text is like this

Oh it's just because the text is running outside of the gLUI "window", isn't it? 😆

`gLUI->window_size = gLUI->splash_size + ImVec2{ 0.0f, 2.0f * ImGui::GetTextLineHeightWithSpacing() };`

That's funny. There's room for a better way to resolve this but scaling the splash size to 90% of the window does still resolve this. Open to suggestions though.

Also for comparison's sake, here is the loading screen example with this PR applied:

![image](https://github.com/user-attachments/assets/9fe96f30-4e57-4c16-9c0a-27bbbbcbdfab)

Reply:
Yea, the window size is based on the image size rather than the screen size. I only did that because centering a window takes less typing than centering something inside a window. Like I said in the original PR, I did the easiest thing.

Reply:
Just a misunderstanding on my part then. I wasn't thinking about the gLUI itself being a window.

Sounds like this should be resolved? Normal stuff, feel free to unmark this resolved or yell if something's wrong, otherwise marking as resolved. And thanks again for the help.

Reply:
You’re welcome!